feature/authentication/src/main/java/com/crisiscleanup/feature/authentication/RootAuthViewModel.kt

```diff
@@ -0,0 +1,44 @@
+package com.crisiscleanup.feature.authentication
+
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.crisiscleanup.core.common.log.AppLogger
+import com.crisiscleanup.core.common.log.CrisisCleanupLoggers
+import com.crisiscleanup.core.common.log.Logger
+import com.crisiscleanup.core.common.network.CrisisCleanupDispatchers
+import com.crisiscleanup.core.common.network.Dispatcher
+import com.crisiscleanup.core.data.repository.AccountDataRepository
+import com.crisiscleanup.core.model.data.AccountData
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.CoroutineDispatcher
+import kotlinx.coroutines.flow.SharingStarted
+import kotlinx.coroutines.flow.map
+import kotlinx.coroutines.flow.stateIn
+import javax.inject.Inject
+
+@HiltViewModel
+class RootAuthViewModel @Inject constructor(
+    private val accountDataRepository: AccountDataRepository,
+    @Dispatcher(CrisisCleanupDispatchers.IO) private val ioDispatcher: CoroutineDispatcher,
+    @Logger(CrisisCleanupLoggers.Auth) private val logger: AppLogger,
+): ViewModel() {
+    val authState = accountDataRepository.accountData
+        .map {
+            if (it.hasAuthenticated) {
```

Comment:
`hasAuthenticated` indicates the user has authenticated at least once. This is true even when the refresh token expires. I'll add documentation later to make things clearer.
Change this to use the `areTokensValid` property. This means the refresh token expiration is in the future and the user won't need to re-authenticate any time soon. So when `areTokensValid = true` the login options don't need to be presented. When `false` the user will want to re-authenticate to get a new refresh token.

Reply:
Got it :+1:

Reply:
Fixed by this commit [`b92e251` (#75)](https://github.com/CrisisCleanup/crisiscleanup-android/pull/75/commits/b92e251040800ae9acada1063629878143d7b2f8)